src/strategies/ens-all-club-digits/index.ts

```diff
@@ -0,0 +1,76 @@
+import { subgraphRequest } from '../../utils';
+import { getAddress } from '@ethersproject/address';
+
+const ENS_SUBGRAPH_URL = {
+  '1': 'https://api.thegraph.com/subgraphs/name/ensdomains/ens',
+  '3': 'https://api.thegraph.com/subgraphs/name/ensdomains/ensropsten',
+  '4': 'https://api.thegraph.com/subgraphs/name/ensdomains/ensrinkeby',
+  '5': 'https://api.thegraph.com/subgraphs/name/ensdomains/ensgoerli'
+};
+
+export const author = '3757';
+export const version = '0.1.1';
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const { numberOfDigits } = options;
+  // 999 Club = 3 , 10k club = 4, 100k club = 5
+  const NAMES_CLUB = [...Array(Math.pow(10, numberOfDigits)).keys()]
+    .map((i) => -1 + i + 1)
+    .map((x) => x.toString().padStart(numberOfDigits, '0'));
+
+  const max = 10;
+  const count = Math.ceil(addresses.length / max);
+  const pages = Array.from(Array(count)).map((x, i) =>
+    addresses.slice(max * i, max * (i + 1))
+  );
+  const params = Object.fromEntries(
+    pages
+      .map((page, i) => `_${i}`)
+      .map((q, i) => [
+        q,
+        {
+          __aliasFor: 'registrations',
+          __args: {
+            block: snapshot !== 'latest' ? { number: snapshot } : null,
+            where: {
+              registrant_in: pages[i].map((address) => address.toLowerCase())
+            },
+            first: 1000,
```

Comment:
Hi @onigiri-x What happens if the registrant has more than 1000 registrations? Maybe we should paginate here

Reply:
Hey Chaitu, this is an excellent question. So what I did is I changed to 4 digit and I found an account with over 1000 ENS names, and just as you predicted the snapshot test broke. I added in a new commit some pagination, I followed loosely how other strategies added pagination. In the end by updating this it is all good. All names should be included.